### PR TITLE
Fix button not clickable

### PR DESCRIPTION
**Pull request by leonie2003**
## Description

Fix edit button on Flashcard and delete button on Card to only be clickable with the respective permissions

## Resolved Issues

resolves part of #120 (delete button is enabled for all users)

### CODE REVIEW

File: frontend/pages/deck/[id]/learn.tsx [hunk 1]
```diff
@@ -21,7 +22,14 @@ export default function Page() {
...
 	const { data: dueCards } = useSWR(`/api/decks/${deckID}/dueCards`, fetcher);
-
+	const { data: deck } = useSWR(
+		deckID ? `/api/decks/${deckID}` : null,
+		fetcher
+	);
+	const { data: group } = useSWR(
+		deck?.groupID ? `/api/groups/${deck.groupID}` : null,
+		fetcher
+	);
```
[the remark this answers is not in the document]
Comment (leonie2003) [reply frontend/pages/deck/[id]/learn.tsx:32]: I will open an issue.

File: frontend/components/flashcard/Card.stories.ts [hunk 2]
```diff
@@ -7,15 +7,7 @@ const meta: Meta<typeof Card> = {
...
-	args: {},
-};
-
-export default meta;
-type Story = StoryObj<typeof Card>;
-
-export const Default: Story = {
-	args: {
-		card: {
+	args: {card: {
```
Comment (Julian702) [inline frontend/components/flashcard/Card.stories.ts:10]: Please run the formatter.

**Review by Julian702 — APPROVED** [review]

**Review by memeToasty — APPROVED** [review]

File: frontend/components/flashcard/Flashcard.stories.ts [hunk 3]
```diff
@@ -49,23 +49,32 @@ export const Back: Story = {
 	},
 };
 
+export const Editable: Story = {
+	args: {
+		editable: true,
+	},
+};
+
```
Comment (JJthere) [inline frontend/components/flashcard/Flashcard.stories.ts:57]: Is this really needed?
Can't you see the same thing in the next three stories?

**Review by JJthere — COMMENTED** [review]
Well done!
Just found two small nits.

File: frontend/components/flashcard/Flashcard.stories.ts (hunk 3 above)
Comment (leonie2003) [reply frontend/components/flashcard/Flashcard.stories.ts:57]: You see different views when you open the stories individually and not in the docs.

**Review by JJthere — APPROVED** [review]

**Review by TomRomeo — APPROVED** [review]

**Review by darmiel — APPROVED** [review]

**Review by 4KevR — APPROVED** [review]

**Pull request merged**